### ISSUE TITLE
Add support for intermediate CAs in kubernetes

### ISSUE THREAD

**Issue by andreas-ahman** [hashicorp/consul#11650]
#### Feature Description

We're using intermediate CA's leveraging Vault PKI, one per DC.
For VM's each VM itself retrieves a signed certificate from the CA and appends the root_CA and intermeditate_CA to the returned certificate since this is required for the different DC's to achieve trust.

We're now looking into adding a k8s cluster as a separated DC to the mesh.
In the helm chart we specify the root_ca and intermediate_ca certificate chain as the CA Cert and then the Private Key for the intermediate CA.

Now consul is able to sign certificates for consul. But since these certificates does not contain the entire trust chain remote servers are unable to verify the authenticity. To work around this we do the following.

1. Retrieve the server cert from the secret `consul-server-cert`
1. Append the root CA and intermediate CA certificates to the certificate
1. Update the secret with the new certificate

So basically we're looking for support for intermediate CA's in kubernetes.

#### Use Case(s)

Use case would be the possibility to use intermediate CA's for kubernetes clusters.

**Comment by dnephin** [comment]
Thank you for opening this issue! This sounds like it might be the same ask as #11598. Does that enhancement seem like it will address the problem for you?

**Comment by andreas-ahman** [comment]
Yes it seems that enhancement should solve our issues as well.
I'll close this issue and will monitor that other issue instead.
